### ISSUE TITLE
sq.gr.co_occurrence gives different results for n_splits=1 and n_splits>1

### ISSUE THREAD

**Issue by JuergenLippoldt** [scverse/squidpy#755]
## Description
Calculating co-occurrences I noticed some inf and nan values in the outcome and while looking into it I saw Issue #689 and tried to reduce n_splits to 1. Not only did the inf and nan values go away, but all the other values changed significantly. 

I would very much appreciate a fix because the function n_splits=1 produces an error for very large samples.
Thank you :)

## Version
1.3.0

**Comment by giovp** [comment]
yes, it's expected, that is because of the way the co-occurrence is implemented, we split the data in chunks if it is too large. I would suggest to split the data by slide and run the co occurrence in each slide separately maybe that helps. It would be cool to have faster implementations but I don't have time to look at this now

**Comment by wenjie1991** [comment]
I'll try to optimize `sq.gr.co_occurrence` to improve its speed. Inspired by the significant speed improvement (4X) in Scanpy's Leiden clustering after using an igraph's implementation, I plan to make a Rust-based backend for the co-occurrence analysis.
